python/tvm/meta_schedule/testing/torchbench/run.py

```diff
@@ -0,0 +1,504 @@
+        """,
+    )
+    args.add_argument(
+        "--adaptive-training",
+        action="store_true",
+        help="Whether to use adpative training for cost model.",
+    )
+    args.add_argument(
+        "--cpu-flush",
+        action="store_true",
+        help="Whether to perform CPU cache flush.",
+    )
+
+    # RPC-related args
+    args.add_argument(
+        "--rpc-host",
+        type=str,
+        help="Host of the RPC Tracker for tuning. Use LocalRunner if not provided",
+    )
+    args.add_argument(
+        "--rpc-port",
+        type=int,
+        help="Port of the RPC Tracker for tuning",
+    )
+    args.add_argument(
+        "--rpc-key",
+        type=str,
+        help="Key of the RPC Tracker for tuning",
+    )
+
+    parsed = args.parse_args()
+    return parsed
+
+
+logging.basicConfig(
+    format="%(asctime)s.%(msecs)03d %(levelname)s %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
+)
+logging.getLogger("tvm.meta_schedule").setLevel(logging.INFO)
+ARGS = parse_args()
+IS_CUDA = ARGS.target.kind.name == "cuda"
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+runner = load_torchdynamo_benchmark_runner(IS_CUDA)
+import torchdynamo
+
+
+def get_metaschedule_runner():
+    if ARGS.rpc_host is not None:
+        return ms.runner.RPCRunner(
+            rpc_config=ms.runner.RPCConfig(
+                tracker_host=ARGS.rpc_host,
+                tracker_port=ARGS.rpc_port,
+                tracker_key=ARGS.rpc_key,
+                session_timeout_sec=600,
+            ),
+            evaluator_config=ms.runner.EvaluatorConfig(
+                number=ARGS.number,
+                repeat=ARGS.repeat,
+                min_repeat_ms=ARGS.min_repeat_ms,
+                enable_cpu_cache_flush=ARGS.cpu_flush,
+            ),
+            alloc_repeat=1,
+        )
+    else:
+        return ms.runner.LocalRunner()
+
+
+def get_tune_config():
+    if ARGS.mode == "eval":
+        max_trials_per_task = 0
+        max_trials_global = 0
+    else:
+        max_trials_per_task = ARGS.max_trials_per_task
+        max_trials_global = ARGS.num_trials
+
+    if max_trials_per_task is None:
+        max_trials_per_task = max_trials_global
+
+    return ms.TuneConfig(
+        strategy="evolutionary",
+        num_trials_per_iter=64,
+        max_trials_per_task=max_trials_per_task,
+        max_trials_global=max_trials_global,
+        adaptive_training=ARGS.adaptive_training,
+    )
+
+
+def get_graph_executor_forward(mod):
+    def forward(*args):
+        if IS_CUDA:
+            torch.cuda.synchronize()
+        args = [arg.contiguous() for arg in args]
+        for idx, arg in enumerate(args, 0):
+            mod.set_input(
+                f"inp_{idx}",
+                tvm.nd.from_dlpack(arg),
+            )
+        mod.run()
+        result = [torch.from_dlpack(mod.get_output(i)) for i in range(mod.get_num_outputs())]
+        if IS_CUDA:
+            torch.cuda.synchronize()
```

Comment:
let's think twice about the synchronization here. two questions:
- `torch.cuda.synchronize` or `tvm.cuda(0).sync()`
- do we want to move the `sync` one line up before `mod.get_output()`?

Reply:
My thoughts are:
1. We should call torch's synchronize before the computation here then call tvm's synchronize afterward, letting each library's sync to block computation from their side.
2. We probably want to move sync before `get_output`. I believe that it doesn't matter as long as `torch.from_dlpack(mod.get_output(i))` is zero-copy. But it doesn't hurt to move sync before that line.